ambry-store/src/main/java/com/github/ambry/store/DiskManager.java

```diff
@@ -570,4 +590,138 @@ private void reportUnrecognizedDirs() {
       }
     }
   }
+
+  public DiskHealthStatus getDiskHealthStatus(){
+    return diskHealthStatus;
+  }
+
+  /**
+    Manages the disk healthchecking tasks such as creating,writing, reading, and deleting a file
+   */
+  private class DiskHealthCheck{
+    private final int operationTimeoutSeconds;      //how long a read/write operation is allotted
+    private final boolean enabled;                  //enabled healthchecking
+    private final String heathcheckDir;             //filepath of where the file will be created
+
+
+    DiskHealthCheck(int operationTimeoutSeconds, boolean enabled, String heathcheckDir){
+      this.enabled = enabled;
+      this.heathcheckDir = heathcheckDir;
+      this.operationTimeoutSeconds = operationTimeoutSeconds;
+
+    }
+
+    /**
+     * Getter utility for enabled boolean
+     * @return flag to indicate disk healthchecking or not
+     */
+    public boolean isEnabled(){
+      return enabled;
+    }
+
+    /**
+     * Attempts to close the filechannel and delete the file
+     * @param fileChannel the file used for disk healthcheck
+     * @return True/False depending on the close/delete was successful
+     */
+    private boolean deleteFile(AsynchronousFileChannel fileChannel){
+      if(fileChannel.isOpen()){
+        try{
+          fileChannel.close();
+        }catch(IOException e){
+          diskHealthStatus = diskHealthStatus.DELETE_IO_EXCEPTION;
+          logger.error("Exception occurred when deleting a file for disk healthcheck", e);
+          return false;
+        }
+      }
+      return true;
+    }
+
+    /**
+     * Performing Disk HealthCheck Test in 5 steps
+     * 1. Checks if there is sufficient space on disk
+     * 2. Creates the file on the filepath
+     * 3. Writes today's date to disk
+     * 4. Read and compares what was written to disk
+     * 5. Deletes the file
+     */
+    public void diskHealthTest(){
+      if(!enabled){ return;} //ensures disk checks are enabled
+
+      //Creates directory for the file if not already there
+      File dir = new File(heathcheckDir);
```

Comment:
please move all disk operations into the try block. This mkdir could also fail if the disk or filesystem is bad

Reply:
I have now. Appreciate it